.github/workflows/ci.yml

```diff
@@ -92,11 +92,13 @@ jobs:
           pip install -U -c constraints.txt -r requirements-dev.txt
       - name: Run unit tests
         run: make unit-test-coverage
-      - name: Upload unit test coverage report
-        uses: actions/upload-artifact@v2
+      - name: Report coverage to coveralls.io
+        uses: coverallsapp/github-action@1.1.3
         with:
-          name: Unit test coverage report - python${{ matrix.python-version }}-${{ matrix.os }}
-          path: htmlcov
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          flag-name: unit-tests_python${{ matrix.python-version }}-${{ matrix.os }}
+          parallel: true
```

Comment:
Given that our coverage differs based on environment, test type and potentially other factors I like the ability to report coverage metrics for all jobs in a single run.

Shows up in Coveralls like below
<img width="1150" alt="image" src="https://user-images.githubusercontent.com/16404496/153221340-b5a09643-8afe-427b-9b8b-fe6505c3b21b.png">